src/relayserver/RelayServer.ts

```diff
@@ -37,10 +37,15 @@ import Timeout = NodeJS.Timeout
 const VERSION = '2.0.0-beta.3'
 const GAS_RESERVE = 100000
 
+// must receive some response for request from node
+const READY_TIMEOUT = 30000
+
 export class RelayServer extends EventEmitter {
   lastScannedBlock = 0
   lastRefreshBlock = 0
   ready = false
+  lastWorkerFinished = -1
+  lastError = null
```

Comment:
I am not sure this is a right approach to error logging, to be honest.

Reply:
Like, "this error happened once and I don't see it happening ever again" is a very different thing from "ok, this error has happened 356834 times over the last minute, something is clearly not good".